File: configs/appconfig.py
# Application Configuration
# Centralized configuration for all ports and endpoints
# import os
from pathlib import Path

class AppConfig:
    """Centralized application configuration"""
    
    USER_DIR = Path(__file__).parent.parent / "user"
    # Port Configuration
    FRONTEND_PORT = 5173
    BACKEND_PORT = 8021
    
    # URLs (derived from ports)
    FRONTEND_URL = f"http://localhost:{FRONTEND_PORT}"
    BACKEND_URL = f"http://localhost:{BACKEND_PORT}"
    
    # API Configuration
    API_PREFIX = "/api"
    
    @classmethod
    def get_frontend_url(cls):
        return cls.FRONTEND_URL
    
    @classmethod
    def get_backend_url(cls):
        return cls.BACKEND_URL
    
    @classmethod
    def get_api_base_url(cls):
        return f"{cls.BACKEND_URL}{cls.API_PREFIX}"

# For backward compatibility and easy imports
FRONTEND_PORT = AppConfig.FRONTEND_PORT
BACKEND_PORT = AppConfig.BACKEND_PORT
FRONTEND_URL = AppConfig.FRONTEND_URL
BACKEND_URL = AppConfig.BACKEND_URL
USER_DIR = AppConfig.USER_DIR

File: configs/appconfig.js
// Application Configuration
// Centralized configuration for all ports and endpoints in JavaScript format
// This can be imported by both frontend and backend (when needed)

export const AppConfig = {
  // Port Configuration
  FRONTEND_PORT: 5173,
  BACKEND_PORT: 8021,

  // URLs (derived from ports)
  get FRONTEND_URL() {
    return `http://localhost:${this.FRONTEND_PORT}`
  },

  get BACKEND_URL() {
    return `http://localhost:${this.BACKEND_PORT}`
  },

  // API Configuration
  API_PREFIX: '/api'
}

// Named exports for convenience
export const FRONTEND_PORT = AppConfig.FRONTEND_PORT
export const BACKEND_PORT = AppConfig.BACKEND_PORT
export const FRONTEND_URL = AppConfig.FRONTEND_URL
export const BACKEND_URL = AppConfig.BACKEND_URL
